### PR TITLE
Add directional skipgram into ModelType

### PR DESCRIPTION
**Pull request by NianhengWu**

### CODE REVIEW

File: finalfrontier/src/skipgram_trainer.rs [hunk 1]
```diff
@@ -186,6 +188,11 @@ where
                 (token * self.ctx_size * 2) + offset
             }
             ModelType::SkipGram => token,
+            ModelType::DirectionalSkipgram => {
+                let offset = if offset_idx < focus_idx { 0 } else { 1 };
+
+                (token * 2) + offset
```
Comment (sebpuetz) [inline finalfrontier/src/skipgram_trainer.rs]: Only some cosmetic change:

```Rust
if offset_idx < focus_idx {
    token * 2
} else {
    token * 2 + 1
}
```

**Comment by sebpuetz** [comment]
Not really anything to complain about

File: finalfrontier/src/skipgram_trainer.rs (hunk 1 above)
Comment (danieldk) [reply finalfrontier/src/skipgram_trainer.rs]: I liked the original formulation ;)

Comment (danieldk) [reply finalfrontier/src/skipgram_trainer.rs]: Though I was surprised that rustfmt would't split this in multiple lines.

Comment (sebpuetz) [reply finalfrontier/src/skipgram_trainer.rs]: Well, I did approve the PR, so we can just ignore my comment ;)

Comment (sebpuetz) [reply finalfrontier/src/skipgram_trainer.rs]: > Though I was surprised that rustfmt would't split this in multiple lines.

I've seen this in other places, too:

```Rust
        let epoch_status = if best_epoch == epoch { "🎉" } else { "" };
```

Taken from sticker!

**Pull request merged**